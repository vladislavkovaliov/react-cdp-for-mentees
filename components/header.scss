@import '../constants/_variables.scss';

.header {
  width: calc(100% - 4px - 4px);
  padding: {
    top: 12px;
    bottom: 48px;
    left: 4px;
    right: 4px;
  };
  background-size: cover;
  background-position: 50%;
  background-image: url("/static/img/netflix-bg.jpg");

  color: #FFFFFF;

  text-transform: uppercase;

  &__title {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin: {
      bottom: 36px;
    }

    & > p {
      text-transform: lowercase;
      color: $red;
      font-size: 12px;
    }

    & > a {
      text-decoration: none;
      width: 78px;
      font-size: 14px;
      border: 1px solid $grey;
      background-color: $white;
      text-align: center;
    }
  }

  &__content {
    &-title {
      font-size: 14px;
    }

    &-input {
      width: calc(100% - 8px - 8px);
      margin-top: 8px;
      padding: {
        left: 8px;
        right: 8px;
      }
      outline: none;
      border: none;
      height: 28px;
      background-color: $black;
      color: $white;
      border-bottom: 1px solid $red;

      &--invalid {
        font-size: 12px;
        color: $white;
        text-transform: lowercase;
      }
    }
  }
}

.header {
  & > div {
    max-width: 980px;
    width: 100%;
    margin: 0 auto;
  }
}

.header__footer,
.header__footer > div {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.header__footer {
  padding-top: 18px;
  padding-bottom: 18px;
  justify-content: space-between;
  font-size: 11px;

  input[type="radio"] {
    display: none;
  }

  label {
    background-color: $grey;
    padding: {
      right: 8px;
      left: 8px;
    }
    margin: {
      left: 4px;
      right: 4px;
    }
    cursor: pointer;
  }

  input[type="radio"]:checked + label {
    background-color: $light-red;
  }

  label,
  button {
    padding: {
      top: 8px;
      bottom: 8px;
    }
  }

  button {
    max-width: 128px;
    width: 100%;
    color: $white;

    text-transform: uppercase;

    background-color: $light-red;
    outline: none;
    border: none;

    cursor: pointer;
  }
}
