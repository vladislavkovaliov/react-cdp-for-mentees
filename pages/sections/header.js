import React, { Component } from 'react';
import { RadioButton } from '../../components/radio-btn.component';
import { connect } from 'react-redux';
import { searchMovies, clearMovies } from '../../actions/movies.action';
import isEmpty from 'lodash/isEmpty';
import { HeaderWrapper } from '../../components/header.component';
import { HeaderTitle } from '../../components/header-title.component';
import { updateSearchingParameters } from '../../actions/searching-parameters.action';
import { InvalidSearchValue } from '../../components/invalid-search-value.component';
import { isValueInputValue } from '../../utils/is-valid-input-value.util';
import {
  DEFAULT_SEARCH_BY,
  EMPTY_STRING,
  MAX_LIMIT_MOVIES,
  DEFAULT_OFFSET,
} from '../../constants/values.constants';

export class HeaderContent extends Component {
  state = {
    search: EMPTY_STRING,
    searchBy: DEFAULT_SEARCH_BY,
    isValid: false,
  };

  onSubmit = (event) => {
    event.preventDefault();

    const { search, searchBy } = this.state;
    const { searchMovies, updateSearchingParameters, clearMovies } = this.props;
    const { sortBy, sortOrder } = this.props.movies;

    if(isEmpty(search) || isEmpty(searchBy)) {
      // alert('You missed search value or did set filter value');
      return;
    }

    clearMovies();
    updateSearchingParameters({ search, searchBy, offset: DEFAULT_OFFSET, limit: MAX_LIMIT_MOVIES });
    searchMovies({ search, searchBy, sortBy, sortOrder, offset: DEFAULT_OFFSET, internalOffset: 0 });
  };

  onChangeInput = (event) => {
    const { searchBy } = this.state;
    const { value: search } = event.target;

    this.setState({ search, searchBy, isValid: isValueInputValue(search) });
  };

  onChangeRadio = (event) => {
    const { value: searchBy } = event.target;

    this.setState({ searchBy });
  };

  render() {
    const { isValid, search, searchBy } = this.state;

    return (
      <HeaderWrapper>
        <form method="#" action="#" onSubmit={this.onSubmit}>
          <HeaderTitle />
          <div className="header__content">
            <div className="header__content-title">find your movie</div>
            <input
              type="text"
              autoFocus
              className="header__content-input"
              onChange={this.onChangeInput}
              value={search}
            />
            <InvalidSearchValue isValid={isValid}>Should be more than 3 symbols</InvalidSearchValue>
          </div>

          <div className="header__footer">
            <div>
              <p>search by</p>
              <RadioButton
                name="filter"
                id="title"
                value="title"
                text="title"
                onChange={this.onChangeRadio}
                isChecked={searchBy === 'title'}
              />
              {/* TODO: should replace to genre */}
              <RadioButton
                name="filter"
                id="genres"
                value="genres"
                text="genres"
                onChange={this.onChangeRadio}
                isChecked={searchBy === 'genres'}
              />
            </div>
            <button id="search-btn" type="submit">search</button>
          </div>
        </form>
      </HeaderWrapper>
    );
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
});

export const mapDispatchToProps = (dispatch) => ({
  searchMovies:              payload => searchMovies(dispatch, payload),
  updateSearchingParameters: payload => updateSearchingParameters(dispatch, payload),
  clearMovies:               payload => clearMovies(dispatch, null),
});

export const Header = connect(mapStateToProps, mapDispatchToProps)(HeaderContent);