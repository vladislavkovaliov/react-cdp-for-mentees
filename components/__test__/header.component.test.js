import React from 'react';
import { HeaderWrapper } from '../header.component';
import { shallow } from 'enzyme';

describe('<HeaderWrapper />', () => {
  describe('Shapshot', () => {
    test('should render <HeaderWrapper />', () => {
      const wrapper = shallow(
        <HeaderWrapper>
          <div>mocks children</div>
        </HeaderWrapper>
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});