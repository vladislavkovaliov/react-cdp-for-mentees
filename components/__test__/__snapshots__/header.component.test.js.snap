// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderWrapper /> Shapshot should render <HeaderWrapper /> 1`] = `
<div
  className="header"
>
  <div>
    <div>
      mocks children
    </div>
  </div>
</div>
`;
